core/state_transition.go

```diff
@@ -684,23 +684,31 @@ func (st *StateTransition) distributeTxFees() error {
 	log.Trace("distributeTxFees", "from", from, "refund", refund, "feeCurrency", st.msg.FeeCurrency,
 		"coinbaseFeeRecipient", st.evm.Context.Coinbase, "coinbaseFee", tipTxFee,
 		"feeHandler", feeHandlerAddress, "communityFundFee", baseTxFee)
+
+	var l1Cost *big.Int
+	// Check that we are post bedrock to enable op-geth to be able to create pseudo pre-bedrock blocks (these are pre-bedrock, but don't follow l2 geth rules)
+	// Note optimismConfig will not be nil if rules.IsOptimismBedrock is true
+	if optimismConfig := st.evm.ChainConfig().Optimism; optimismConfig != nil &&
+		st.evm.ChainConfig().IsOptimismBedrock(st.evm.Context.BlockNumber) {
+		l1Cost = st.evm.Context.L1CostFunc(st.evm.Context.BlockNumber.Uint64(), st.evm.Context.Time, st.msg.RollupDataGas, st.msg.IsDepositTx)
+	}
+
 	if feeCurrency == nil {
 		st.state.AddBalance(st.evm.Context.Coinbase, tipTxFee)
 		st.state.AddBalance(from, refund)
 
-		// Check that we are post bedrock to enable op-geth to be able to create pseudo pre-bedrock blocks (these are pre-bedrock, but don't follow l2 geth rules)
-		// Note optimismConfig will not be nil if rules.IsOptimismBedrock is true
-		if optimismConfig := st.evm.ChainConfig().Optimism; optimismConfig != nil &&
-			st.evm.ChainConfig().IsOptimismBedrock(st.evm.Context.BlockNumber) &&
-			st.evm.ChainConfig().IsCel2(st.evm.Context.Time) {
+		if st.evm.ChainConfig().IsCel2(st.evm.Context.Time) {
 			st.state.AddBalance(feeHandlerAddress, baseTxFee)
-			if cost := st.evm.Context.L1CostFunc(st.evm.Context.BlockNumber.Uint64(), st.evm.Context.Time, st.msg.RollupDataGas, st.msg.IsDepositTx); cost != nil {
-				st.state.AddBalance(params.OptimismL1FeeRecipient, cost)
-			}
+		}
+
+		if st.evm.ChainConfig().IsOptimismBedrock(st.evm.Context.BlockNumber) && l1Cost != nil {
```

Comment:
Since `l1Cost` is [conditional](https://github.com/celo-org/op-geth/pull/35/files#diff-8c398ef0a79f97ba6d497a99247a815b3a51918de9e16954215b6073f907c92eR692), does it make sense to reduce to just `l1Cost != nil` like [here](https://github.com/celo-org/op-geth/pull/35/files#diff-8c398ef0a79f97ba6d497a99247a815b3a51918de9e16954215b6073f907c92eR708)

Reply:
Yes, good catch!